cli/src/lib.rs

```diff
@@ -138,6 +138,14 @@ where
         .await
     }
 
+    fn prepare_panic_hook(notify_shutdown: Arc<Notify>) {
+        let hook = panic::take_hook();
+        panic::set_hook(Box::new(move |info| {
+            hook(info);
+            notify_shutdown.notify_one();
```

Comment:
I might have missed some detail, but wouldn't it require
[`notify_waiters()`](https://docs.rs/tokio/latest/tokio/sync/struct.Notify.html#method.notify_waiters) 
to send the signal to both the telemetry and API servers? In the same way it was done
in the `cli::start_listening_signal()`, for example.

Reply:
Oh, thanks!

Reply:
Hm, with `notify_waiters` it doesn't pass the test. The notification keeps being awaiting.

Reply:
Then I guess it's fine as it is. Maybe asking @pesterev to take a look would
make sense, as he did the other `Notify` PR, but I'll approve for now.